src/solvers/flattening/arrays.cpp

```diff
@@ -209,24 +211,23 @@ void arrayst::collect_arrays(const exprt &a)
   }
   else if(a.id()==ID_typecast)
   {
-    // cast between array types?
-    DATA_INVARIANT(
-      a.operands().size()==1,
-      "typecast must have one operand");
+    const auto &typecast_op = to_typecast_expr(a);
```

Comment:
I'd expect `typecast_op` to be `to_typecast_expr(a).op()`. Just `typecast_expr`?